packages/core/src/browser/tree/tree-model.ts

```diff
@@ -157,11 +157,7 @@ export class TreeModelImpl implements TreeModel, SelectionProvider<ReadonlyArray
         this.toDispose.push(this.expansionService);
         this.toDispose.push(this.expansionService.onExpansionChanged(node => {
             this.fireChanged();
-            if (!node.expanded && [...this.selectedNodes].some(selectedNode => CompositeTreeNode.isAncestor(node, selectedNode))) {
-                if (SelectableTreeNode.isVisible(node)) {
-                    this.selectNode(node);
-                }
-            }
+            this.selectOnExpansion(node);
```

Comment:
Would `onExpansion` make more sense?

Reply:
It is naming convention for events or notification usually to start `on`. I prefer to give names to clarify intention. We could name `handleExpansion` but it does not explain what method does. I would usually then introduce 2 template methods:
- generic tempalte method `handleExpansion`, so one can override it
- and `selectOnExpansion` which is called by default from `handleExpansion`

Reply:
Just updated ;)